pkg/controller/installation/core_controller.go

```diff
@@ -1289,18 +1310,27 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 				rcertificatemanagement.NewKeyPairOption(typhaNodeTLS.NodeSecret, true, true),
 				rcertificatemanagement.NewKeyPairOption(nodePrometheusTLS, true, true),
 				rcertificatemanagement.NewKeyPairOption(typhaNodeTLS.TyphaSecret, true, true),
+				rcertificatemanagement.NewKeyPairOption(typhaNodeTLS.TyphaSecretNonClusterHost, true, true),
 				rcertificatemanagement.NewKeyPairOption(kubeControllerTLS, true, true),
 			},
 			TrustedBundle: typhaNodeTLS.TrustedBundle,
 		}))
 
+	// Check if non-cluster host feature is enabled.
+	nonclusterhost, err := utils.GetNonClusterHost(ctx, r.client)
```

Comment:
Should only happen if installation.spec.variant == enterprise

Reply:
Nice catch. I have added this check for the enterprise variant.